File: docs/notintoc/codesnippet/VisualBasic/walkthrough--autoloading-toolbox-items_9.vb
    Protected Overrides Sub Initialize()
        Trace.WriteLine(String.Format(CultureInfo.CurrentCulture, _
            "Entering Initialize() of: {0}", Me.GetType().Name))
        MyBase.Initialize()

        ' Add our command handlers for menu (commands must exist in the .vsct file)
        Dim mcs As OleMenuCommandService = _
            TryCast(GetService(GetType(IMenuCommandService)), OleMenuCommandService)
        If Not mcs Is Nothing Then

            ' Create the command for the menu item.
            Dim menuCommandID As New CommandID( _
                GuidList.guidLoadToolboxMembersCmdSet, CInt(PkgCmdIDList.cmdidMyCommand))
            Dim menuItem As New MenuCommand( _
                New EventHandler(AddressOf MenuItemCallback), menuCommandID)
            mcs.AddCommand(menuItem)
        End If

        ' Use reflection to get the toolbox items provided in this assembly.
        ToolboxItemList = CreateItemList(Me.GetType().Assembly)
        If ToolboxItemList Is Nothing Then
            ' Unable to generate the list.
            ' Add error handling code here.
        End If
    End Sub

File: docs-old-1102/misc/codesnippet/CSharp/walkthrough-autoloading-toolbox-items_9.cs
        protected override void Initialize()
        {
            Trace.WriteLine (string.Format(CultureInfo.CurrentCulture,
                "Entering Initialize() of: {0}", this.ToString()));
            base.Initialize();

            // Add our command handlers for menu (commands must exist in the .vsct file)
            OleMenuCommandService mcs =
                GetService(typeof(IMenuCommandService)) as OleMenuCommandService;
            if ( null != mcs )
            {
                // Create the command for the menu item.
                CommandID menuCommandID =
                    new CommandID(GuidList.guidLoadToolboxMembersCmdSet,
                        (int)PkgCmdIDList.cmdidMyCommand);
                MenuCommand menuItem =
                    new MenuCommand(MenuItemCallback, menuCommandID );
                mcs.AddCommand( menuItem );

                // Subscribe to the toolbox intitialized and upgraded events.
                ToolboxInitialized += new EventHandler(OnRefreshToolbox);
                ToolboxUpgraded += new EventHandler(OnRefreshToolbox);
            }

            // Use reflection to get the toolbox items provided in this assembly.
            ToolboxItemList = CreateItemList(this.GetType().Assembly);
            if (null == ToolboxItemList)
            {
                // Unable to generate the list.
                // Add error handling code here.
            }
        }